cycling_utils/sampler.py

```diff
@@ -113,3 +115,170 @@ def in_epoch(self, epoch):
         self.set_epoch(epoch)
         yield
         self._reset_progress()
+
```

Comment:
Why does the default interruptable sampler not handle batching - You can just batch your dataset first?

 Wouldn't it just be better to provide a more generic base class that you can pass a batch_by or group_by with some generic function?

Reply:
Batching with the interruptible sampler might be possible with some modification. 

The IDGBS takes a list of batch labels as inputs already, leaving that to the user to generate, then after randomly sharding the data across the available GPUs it groups the data per GPU into batches based on batch labels. We could chain these batches into one stream again and yield from that?